### PR TITLE
Add support for newer openshift labels in test infra

### LINKED ISSUE #302: Pipelines run on OCP 4.6
## Description
As an end user who runs the Secrets Provider in OpenShift,
I want the Secrets Provider to run on OCP 4.6
So that I can use the entire DAP product on my 4.6 cluster

## Details
To implement this, we want to add a pipeline flow that runs the `next` cluster on a pipeline trigger.
- [x] Update deploy/summon/secrets.yml to reference `oldest`, `current`, and `next` with no hardcoded versions (version can be taken from the conjurops var `ACCOUNT/openshift/CLUSTER/version`
- [x] Update existing flows to remove hardcoded OCP versions and instead reference `oldest` and `current` / use the OCP version value from ConjurOps (stored in the `OPENSHIFT_VERSION` env var). Note that Secrets Provider test flows should NOT run against `oldest` in this update - the `oldest` flow (OCP 4.3) is currently not supported.
- [x] Add a flow to the Jenkinsfile for `next`:
       - Add a boolean parameter for whether to run the `next` pipeline (this should only be run when testing newer OCP versions):
         ```
          parameters {
            booleanParam(name: 'TEST_OCP_NEXT', description: 'Whether or not to run the pipeline against the next OCP version', defaultValue: false)
          }
          ```
        - Add a stage for testing against `next` that is gated on `when { expression { params.TEST_OCP_NEXT } }`

Once these changes are in place, we can test flows against the `next` cluster by:
- [x] Run the pipeline against the `next` cluster by setting the `TEST_OCP_NEXT` value to `true` in the Jenkins UI

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
### What does this PR do?
We now support running tests against the 'next', 'current', and 'oldest' versions of Openshift, as defined in Conjurops.

Currently, only 'current' and 'next' are supported

### What ticket does this PR close?
Resolves #302 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [X] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: Jenkinsfile
Comment (orenbm) [inline Jenkinsfile:116]: thanks @BradleyBoutcher for getting this!
Can we verify that the OCP version is logged in a clear place so we know which versions are tested in each Jenkins run?

**Review by orenbm — DISMISSED** [review]
Thanks @BradleyBoutcher !

Just blocking this PR from being merged just so we don't merge the OSS tests removal by accident :)

File: Jenkinsfile
Comment (izgeri) [inline Jenkinsfile:112]: we will need to keep the 3.11 tests in here for now

File: Jenkinsfile
Comment (izgeri) [inline Jenkinsfile:115]: maybe we should add a test flow against `oldest` but comment it out for now? oldest is currently 4.3, and SP is not certified for 4.3

File: bin/start
Comment (diverdane) [inline bin/start:23]: Silly nit: The spacing before the word `Run` is off by 1 for `--oldest` and `--next` options.
That's all I've got, everything else looks good to me (beyond comments already made).

File: bin/start
Comment (izgeri) [inline bin/start:37]: nit, but if we add this here let's use the long form of the flag `--provider`

File: deploy/run_with_summon.sh
Comment (izgeri) [inline deploy/run_with_summon.sh:25]: can we use `announce` instead of echo? it will def get lost in the logs with just echo.

File: deploy/summon/secrets.yml
Comment (izgeri) [inline deploy/summon/secrets.yml:51]: it looks like kubernetes-conjur-deploy expects this to be `openshift*` (though I'm not sure why it's working now)
https://github.com/cyberark/kubernetes-conjur-deploy/blob/fbe14df8a20e451ffdebca0d4b73c85df52d2bcc/test.sh#L43-L51

you can probably just make them all the literal strings `openshift` instead
```suggestion
  TEST_PLATFORM: openshift
```

File: deploy/summon/secrets.yml
Comment (izgeri) [inline deploy/summon/secrets.yml:66]:
```suggestion
  TEST_PLATFORM: openshift
```

File: deploy/summon/secrets.yml
Comment (izgeri) [inline deploy/summon/secrets.yml:81]:
```suggestion
  TEST_PLATFORM: openshift
```

**Review by izgeri — CHANGES_REQUESTED** [review]
A few minor comments. Also, we should be adding a CHANGELOG entry about validing the secrets provider on OpenShift 4.6.

File: deploy/summon/secrets.yml
Comment (izgeri) [reply deploy/summon/secrets.yml:51]: btw, this seems silly and redundant now. makes me think we should just change the logic in k8s-deploy to use `PLATFORM`, remove `TEST_PLATFORM` from that project, and then remove it from here.

**Review by izgeri — COMMENTED** [review]
LGTM, but if we can verify that `next` is passing we should update the [supported platforms](https://github.com/cyberark/secrets-provider-for-k8s/#supported-platforms) in the README with this PR to include 4.6

**Review by izgeri — APPROVED** [review]
LGTM pending green build, including with `next` pipeline

**Comment by izgeri** [comment]
It's all green, including on OCP 4.6! See [Jenkins](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--secrets-provider-for-k8s/detail/add-openshift-envs/27/pipeline/) for confirmation. Thanks @BradleyBoutcher !!

**Pull request merged**
